data-es6.js

```diff
@@ -10540,12 +10543,12 @@ exports.tests = [
         set.add(123);
         set.add(123);
 
-        return set.has(123);
+        return set instanceof S && set.has(123);
```

Comment:
this should probably check `set instanceof S && s instanceof Set`.

However, to truly be sure it's working properly, `instanceof` is irrelevant - you need to check for internal slots. `return Set.prototype.size.call(set) === 1;` will throw unless it's properly subclassed, ie, the internal slots are set up properly.

I think a similar approach should be taken on all of the builtin subclassing tests.